File: cinnamon-dynamic-wallpaper@TobiZog/files/cinnamon-dynamic-wallpaper@TobiZog/5.4/scripts/location.js
/**
 * @name	Cinnamon-Dynamic-Wallpaper
 * @alias 	TobiZog
 * @since	2023
 * 
 * @description Functions to estimate the user location
 */

/******************** Imports ********************/

const Soup = imports.gi.Soup;


/******************** Functions ********************/

/**
 * Estimate the location of the user
 * 
 * @returns Location data if succeded or -1 if failed
 */
function estimateLocation() {
	let sessionSync = new Soup.SessionSync();
	let msg = Soup.Message.new('GET', "https://get.geojs.io/v1/ip/geo.json");
	sessionSync.send_message(msg);

	if (msg.status_code == 200) {
		return JSON.parse(msg.response_body.data);
	} else {
		return -1;
	}
}

File: cinnamon-dynamic-wallpaper@TobiZog/files/cinnamon-dynamic-wallpaper@TobiZog/5.4/scripts/location.py
import urllib.request, json

class Location():
  """ Class to handle location requests
  """
  def __init__(self):
    self.GEO_URL = "https://get.geojs.io/v1/ip/geo.json"

  def get_location(self) -> dict:
    """ Request the location via network

    Returns:
        dict: latitude and longitude
    """
    request = urllib.request.urlopen(self.GEO_URL)

    data = json.load(request)

    return {
      "latitude": data["latitude"],
      "longitude": data["longitude"]
    }
